_posts/2020-01-30-new-age-of-bastion.md

```diff
@@ -0,0 +1,165 @@
+    .with_actor_restart_strategy(ActorRestartStrategy::LinearBackOff {
+      timeout: Duration::from_secs(1),
+    });
+
+// Define the supervisor
+supervisor
+	 // That uses our restart strategy defined earlier
+	 .with_restart_strategy(restart_strategy)
+	 // And tracks the child group, defined in the following function
+	 .children(|children| failed_actors_group(children))
+```
+1- Now we have a signature method to address all the actors in the system. Check our context addressing system [here](https://docs.rs/bastion/0.3.4/bastion/context/struct.BastionContext.html#method.signature)
+
+```rust
+Bastion::children(|children| {
+    children.with_exec(|ctx: BastionContext| {
+        async move {
+            ctx.tell(&ctx.signature(), "Hello to myself");
+
+            Ok(())
+        }
+    })
+}).expect("Couldn't create the children group.");
+```
+
+2- We've added meta-architecture instantiation. What is that? So now you can have a boilerplate free hierarchy for your actors and/or workers.
+```rust
+let children = children! {
+	// the default redundancy is 1
+	redundancy: 100,
+	action: |msg| {
+	   // do something with the message here
+	},
+};
+
+// simpler supervisor declaration.
+let sp = supervisor! {};
+
+let children = children! {
+  	supervisor: sp,
+  	redundancy: 10,
+  	action: |msg| {
+    	// do something with the message here
+    },
+};
+```
+
+3- You can mark your blocking code and execute that code inside blocking adaptive thread pool of Bastion.
+```rust
+let task = blocking! {
+	thread::sleep(time::Duration::from_millis(3000));
+};
+```
+
+4- What other frameworks and implementation calls `block_on` we call it `run`.  Because that's how other ecosystems call it. We believe general conventions should be over the specific conventions. If you want to run a process and wait for it's completion. (e.g. the blocking task written above) use:
+```rust
+run!(task);
+```
+And that's all now everything will be run. This is equivalent of [zio running effects](https://zio.dev/docs/overview/overview_running_effects) in Rust environment. Now we have it!
+
+-------
+
+**Executor, our huge steampunk engine**
+
+That the piece that we are running everything on top. It is way more diverged and different from any runtime out there. Working with different techniques that we will explore in the upcoming posts.
+
+But what has changed? Let's explore!
+
+0- Under the contention we skip work-stealing. We directly force queue consumer to consume message from the run queue and continue working without work-stealing for a small fraction of time. Think like this is a congestion unblocking mechanism. Works pretty well.
+
+1- Bastion executor relies mostly on interconnect messages and tries to do single clock advancement without OS migration of threads. We are pinning our threads so we need to know how many vCPU is going on and maintain thread pinning on the fly. In this release, we have improved pinning API a lot.
```

Comment:
We are pinning our threads so we need to know how many vCPU is going on and maintain thread pinning on the fly. In this release, we have improved pinning API a lot.

We are pinning our threads which means we need to be aware of how many vCPU Bastion is running on, and we maintain thread pinning on the fly. A lot of efforts were put into our pinning API in this release.